git/gogit/clone.go

```diff
@@ -189,8 +190,39 @@ func (g *Client) cloneTag(ctx context.Context, url, tag string, opts repository.
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve commit object for HEAD '%s': %w", head.Hash(), err)
 	}
+	commit, err := buildCommitWithRef(cc, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	tagRef, err := repo.Tag(tag)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find reference for tag '%s': %w", tag, err)
+	}
+
+	annotated := true
+	tagObj, err := repo.TagObject(tagRef.Hash())
+	if err != nil {
+		// If the tag ref exists but a tag object for the ref does not exist,
+		// treat the tag as a lightweight tag.
+		if err == plumbing.ErrObjectNotFound {
+			annotated = false
+		} else {
+			return nil, fmt.Errorf("unable to resolve tag object for tag '%s' with hash '%s': %w", tag, tagRef.Hash(), err)
+		}
+	}
+	// The commit is considered to have a parent only if the tag is annotated,
+	// as lightweight tags are essentially named pointers.
+	if annotated {
+		annotatedTag, err := buildTag(tagObj)
+		if err != nil {
+			return nil, err
+		}
+		commit.Parent = annotatedTag
```

Comment:
we'd need to pass the repo object to this wrapper since we need to resolve the tag and fetch the tag object.
we also resolve the commit in the clone function and then pass the commit object to `buildCommitWithRef()` to build our repr of the commit, so whats the issue with resolving the tag in the function responsible for cloning tags and then building our repr of a tag using `buildTag()`?